data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,204 @@
+#Usage and help info for the script
+function help_usage()
+{
+        echo "script usage: $(basename $0) [-s Start Monitor for guest console] [-e End of Monitor for guest console ] [-h help]"
+}
+
+#Quit if there are less than six arguments
+if [ $# -eq 0 ];then
+        help_usage
+        exit 1
+fi
+
+#PREPARATION
+##CONFIG PART
+START_MONITOR="0"
+END_MONITOR="0"
+LOG_DIR="/tmp/virt_logs_residence"
+MONITOR_LOCKED_FILE="/tmp/GUEST_CONSOLE_COLLECTOR_SEMAPHORE"
+timestamp="`date '+%Y%m%d%H%M%S'`"
+###CHECK EXISTED GUEST
+vm_guestnames_types="sles"
+get_vm_guestnames_inactive=`virsh list --inactive | grep -E "${vm_guestnames_types}" | awk '{print $2}'`
+vm_guestnames_inactive_array=$(echo -e ${get_vm_guestnames_inactive})
+get_vm_guestnames=`virsh list  --all | grep -E "${vm_guestnames_types}" | awk '{print $2}'`
+vm_guestnames_array=$(echo -e ${get_vm_guestnames})
+vmguest=""
+vmguest_failed="0"
+
+#Parse input arguments. -s or -e must have values
+while getopts "seh" OPTION; do
+  case "${OPTION}" in
+    s)
+      START_MONITOR="1"
+      ;;
+    e)
+      END_MONITOR="1"
+      ;;
+    h)
+      help_usage
+      exit 1
+      ;;
+    *)
+      help_usage
+      exit 1
+      ;;
+  esac
+done
+
+#FUNCTION PART
+function get_console()
+{
+
+local vmguest=$1
+
+expect -c " 
+set timeout -1
+
+#hide echo
+log_user 0
+spawn -noecho virsh console ${vmguest}
+
+#wait connection
+sleep 3
+send \"\r\n\r\n\r\n\"
+
+#condition expect
+expect {
+unset expect_out(buffer)
+        \"*login:\" {
+                send \"root\r\"
+                exp_continue
+        }
+        \"*assword\" {
+                send \"novell\r\"
+                exp_continue
+        }
+        \"*:~ #\" {
+                send -- \"ip route get 1\r\"
+        }
+}
+
+#submatch for output
+expect -re {dev.*\s([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})[^0-9]}
+
+if {![info exists expect_out(1,string)]} {
+        puts \"Match did not happen :(\"
+        exit 1
+}
+
+# assign submatch to variable
+set output \$expect_out(buffer)
+#terminate the current virsh console situation
+send \"\\035\\r\"
+expect eof
+
+puts \"\$output\"
+"
+
+}
+
+
+function open_console()
+{
+        #open guest console
+        while :
+        do
+            for vmguest in ${vm_guestnames_state_running_array[@]};do
+                get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+            done
+            wait
+        done  
+}
+
+function close_console()
+{
+        set -x
+        #close guest console
+        pid=`ps aux | grep "$(basename $0)" | grep -v "grep"| awk '{print $2}' | head -n1`
+        #kill -9 $pid >> /dev/null 2>&1
+        echo $pid
+        kill -9 $pid
+}
+
+function start_monitor()
+{
+        #start monitor for guest console
+        get_vm_guestnames_state_running=`virsh list --state-running | grep -E "${vm_guestnames_types}" | awk '{print $2}'`
+        vm_guestnames_state_running_array=$(echo -e ${get_vm_guestnames_state_running})  
```

Comment:
Can be further simplified to variable=backtick bash command backtick. But not compulsory.

Reply:
Got it